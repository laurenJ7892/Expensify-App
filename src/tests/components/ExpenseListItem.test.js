import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'; 
import selectExpensesTotal from '../../Selectors/expenses-total';

// test('should render test item from expenses', () => {
//     const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
//     expect(wrapper).toMatchSnapshot();
// });

test('should return 0 from expenses total', () => {
    const total = selectExpensesTotal([]);
    expect(total).toBe(0);
});

test('should return total for one expense', () => {
    const total = selectExpensesTotal([expenses[0]]);
    expect(total).toBe(195);
});

test('should return total for multiple expense', () => {
    const total = selectExpensesTotal(expenses);
    expect(total).toBe(530195);
});