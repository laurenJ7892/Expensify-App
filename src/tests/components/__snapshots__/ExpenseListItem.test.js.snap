// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render test item from expenses 1`] = `
<div>
  <[object Object]
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </[object Object]>
  <p>
     
    500000
     - 
    -345600000
     
  </p>
</div>
`;
